pvanalytics/system.py

```diff
@@ -1,10 +1,283 @@
+    clipping : Series
+        Boolean Series identifying where power or irradiance is being
+        clipped.
+    clip_max : float, default 0.1
+        If the fraction of data flagged as clipped is greater than
+        `clip_max` then it cannot be determined whether the system is
+        tracking or fixed and :py:const:`Tracker.UNKNOWN` is returned.
+    envelope_quantile : float, default 0.995
+        Quantile used to determine the upper power or irradiance
+        envelope.
+    fit_median : boolean, default True
+        Perform a secondary fit with the median power or irradiance to
+        validate that the profile is consistent through the entire
+        data set.
+    median_r2_min : float, default 0.9
+        Minimum :math:`r^2` for a curve fit to the median power or
+        irradiance at each minute of the day (Applies only if
+        `fit_median` is True).
+    fit_params : dict or None, default None
+        Minimum r-squared for curve fits according to the fraction of
+        data with clipping. This should be a dictionary with tuple
+        keys and dictionary values. The key must be a 2-tuple of
+        ``(clipping_min, clipping_max)`` where the values specify the
+        minimum and maximum fraction of data with clipping for which
+        the associated fit parameters are applicable. The values of
+        the dictionary are themselves dictionaries with keys
+        ``'fixed'`` and ``'tracking'``, which give the minimum
+        :math:`r^2` for the curve fits, and ``'fixed_max'`` which
+        gives the maximum :math:`r^2` for a quadratic fit if the
+        system appears to have a tracker.
+
+        If None :py:data:`PVFLEETS_FIT_PARAMS` is used.
+    envelope_min_fraction : float, default 0.05
+        After calucluating the power or irradiance envelope data less
+        than `envelope_min_fraction` times the maximum of the envelope
+        is removed. This excludes data from morning and evening that
+        may interfere with curve fitting.
+    median_min_fraction : float, default 0.025
+        After calculating the median power or irradiance at each
+        minute, data less than `median_min_fraction` times the maximum
+        is removed. This excludes data from morning and evening that
+        may interfere with curve fitting.
+    seasonal_split : tuple of list of int, optional
+        Tuple specifying a set of winter months and a set of summer
+        months. The order is not important. The months are specified
+        as integers between 1 and 12. If not specified defaults to
+        ([5, 6, 7, 8], [11, 12, 1, 2]) which works well for North
+        America. Data is split in to two groups, one for each set and
+        curve fits are applied to the upper envelope of each group
+        independently. If the curve fits produce different results
+        (e.g. one TRACKING and one FIXED) then
+        :py:const:`Tracker.UNKNOWN` is returned.
```

Comment:
This one is a bit confusing. Defaults to None but None signals a different default. So it appears that None doesn't turn the seasonal split off, it's a shortcut to the actual default.

Is there a way to turn seasonal splitting off? If yes, then leave the default at None, and describe ([5, ...]) etc. as values that work well in the northern hemisphere.  Otherwise I think it would be better to default to the tuple in the comment and put that default value in the docstring in place of 'optional'.

If we have an option to turn seasonal splitting off, it should be tested.

Reply:
@cwhanse I changed the docstring, but I think we should leave the actual default value as `None` to avoid problems with mutable defaults.

Reply:
How about tuple of tuple instead of tuple of list for mutability concerns?

Reply:
I think I'm unclear on the immutability of a tuple when it holds mutable data. While you can't assign a new object to the elements of a tuple you can still mutate the objects that it already contains
```python
x = ([1, 2], [3])
x[1].append(4)
```
results in `x = ([1, 2], [3, 4])`.

As far as I can tell, that results in the same problem as having a bare list as the default.

Reply:
After reading it again, my previous comment reminds me of the [buffalo sentence](https://en.wikipedia.org/wiki/Buffalo_buffalo_Buffalo_buffalo_buffalo_buffalo_Buffalo_buffalo).

To clarify, I mean using `((5, 6, 7, 8), (11, 12, 1, 2))` as the default value instead of `([5, 6, 7, 8], [11, 12, 1, 2])`.

Reply:
My mistake. that would work! Thanks for your advice, @kanderso-nrel, I'm not used to thinking about tuples in the laid-back way that python treats them.

Reply:
I'm not used to parsing the buffalo sentence....

Thanks @kanderso-nrel for the help and the Friday distraction :)